Components/Operation/Settings.php

```diff
@@ -0,0 +1,87 @@
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * @author Nosto Solutions Ltd <shopware@nosto.com>
+ * @copyright Copyright (c) 2018 Nosto Solutions Ltd (http://www.nosto.com)
+ * @license http://opensource.org/licenses/BSD-3-Clause BSD 3-Clause
+ */
+
+use Shopware\Models\Shop\Shop;
+use Nosto\Operation\UpdateSettings as NostoUpdateSettings;
+use Nosto\Object\Settings as NostoSettings;
+use Shopware_Plugins_Frontend_NostoTagging_Components_Account as NostoComponentAccount;
+use Shopware_Plugins_Frontend_NostoTagging_Bootstrap as Bootstrap;
+use Shopware_Plugins_Frontend_NostoTagging_Components_Helper_Currency as CurrencyHelper;
+use Shopware_Plugins_Frontend_NostoTagging_Components_Operation_ExchangeRates as ExchangeRatesOperation;
+
+/**
+ * Settings operation component. Used for updating settings events to Nosto
+ *
+ * @package Shopware
+ * @subpackage Plugins_Frontend
+ */
+class Shopware_Plugins_Frontend_NostoTagging_Components_Operation_Settings
+{
+    /**
+     * Send an updated settings object with the currency changes
+     *
+     * @param Shop $shop
+     */
+    public static function updateCurrencySettings(Shop $shop)
+    {
+        $settings = new NostoSettings();
+        /** @noinspection PhpUndefinedMethodInspection */
+        $shopConfig = Shopware()->Plugins()->Frontend()->NostoTagging()->getShopConfig($shop);
+        if ($shopConfig[Bootstrap::CONFIG_MULTI_CURRENCY] === Bootstrap::CONFIG_MULTI_CURRENCY_EXCHANGE_RATES) {
+            $settings->setUseCurrencyExchangeRates(true);
+            $defaultCurrency = CurrencyHelper::getDefaultCurrency($shop);
+            if ($defaultCurrency) {
+                $settings->setDefaultVariantId($defaultCurrency->getCurrency());
+            }
+            $settings->setCurrencies(CurrencyHelper::getFormattedCurrencies($shop));
+        } else {
+            $settings->setUseCurrencyExchangeRates(false);
+            $settings->setDefaultVariantId('');
+        }
+        $account = NostoComponentAccount::findAccount($shop);
+        if ($account) {
+            $nostoAccount = NostoComponentAccount::convertToNostoAccount($account);
+            $service = new NostoUpdateSettings($nostoAccount);
+            try {
+                ExchangeRatesOperation::updateCurrencyExchangeRates($nostoAccount, $shop);
```

Comment:
Why is this a static?

Reply:
No Reason to be static. Fixed, thanks :)